programs/leap-util/actions/blocklog.hpp

```diff
@@ -0,0 +1,49 @@
+#include "subcommand.hpp"
+#include <boost/filesystem.hpp>
+#include <boost/filesystem/path.hpp>
+#include <eosio/chain/block_log.hpp>
+#include <eosio/chain/config.hpp>
+
+namespace bfs = boost::filesystem;
+using namespace eosio::chain;
+
+struct blocklog_options {
+   std::string blocks_dir = "blocks";
+   std::string output_file = "";
+   uint32_t first_block = 0;
+   uint32_t last_block = std::numeric_limits<uint32_t>::max();
+   std::string output_dir;
+
+   // flags
+   bool no_pretty_print = false;
+   bool as_json_array = false;
+
+   // subcommands
+   bool make_index;
+   bool trim_blocklog;
+   bool extract_blocks;
+   bool smoke_test;
+   bool vacuum;
+   bool genesis;
```

Comment:
I think you should initialize these all to `false`. Some are set in callbacks which if not called leaves them un-initialized causing undefined behavior.

Reply:
Thank you, good suggestion, done!